source/extensions/filters/network/wasm/wasm_filter.h

```diff
@@ -33,11 +34,12 @@ class FilterConfig : Logger::Loggable<Logger::Id::wasm> {
       return nullptr;
     }
     if (wasm && !root_context_id_) {
-      root_context_id_ = wasm->getRootContext(plugin_->root_id_)->id();
+      root_context_id_ = wasm->getRootContext(plugin_, false)->id();
```

Comment:
is there any possibility that `getRootContext` returns nullptr? What if the same plugin is in `pending_done_` state (though I'm not sure if there's such a case)

Reply:
`Plugin`'s shutdown process starts when `FilterConfig` is destroyed, at which point Envoy won't be calling `createFilter()` to create a new stream, which means that this will be called only for active `RootContext`.

Reply:
thanks! I got it